.github/workflows/pre-release.yaml

```diff
@@ -0,0 +1,65 @@
+# Workflow to fetch the latest commit hash on the main branch and upload artifacts to CF storage.
+name: Build and upload from main
+on:
+  workflow_dispatch:
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  define-matrix:
+    name: Define Matrix
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+    outputs:
+      releases: ${{ steps.prepare-matrix.outputs.releases }}
+      commit_hash: ${{ steps.get-commit-hash.outputs.hash }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: sudo apt-get install -y jq
+
+      - name: Get latest commit hash
+        id: get-commit-hash
+        run: echo "hash=$(git rev-parse --short HEAD)" >> "$GITHUB_OUTPUT"
+
+      - name: Prepare JSON output for matrix
+        id: prepare-matrix
+        run: |
+          RELEASES_JSON=$(find contracts -maxdepth 1 -mindepth 1 -type d | sed 's|contracts/||' | jq -R . | jq -s --arg commit "${{ steps.get-commit-hash.outputs.hash }}" 'map({
+            package_name: .,
+            version: $commit,
+            package_git_tag: "\(.)_\($commit)"
+          })')
```

Comment:
this is overkill, the commit is the same now, we just need to list all contract names

Reply:
But we aren't releasing to crates.io here. Take a look at how the `version` and `package_git_tag` are being used below. The commit hash can be used there directly instead of creating this confusing JSON